PostHog/PostHogStorage.swift

```diff
@@ -122,39 +160,51 @@ class PostHogStorage {
+        guard fileManager.fileExists(atPath: sourceUrl.path) else { return }
+        // Copy file or directory over (if it doesn't exist)
+        if !fileManager.fileExists(atPath: destinationUrl.path) {
+            try fileManager.copyItem(at: sourceUrl, to: destinationUrl)
+        }
+    }
+
     private static func getAppFolderUrl(from configuration: PostHogConfig) -> URL {
-        /**
-
-         From Apple Docs:
-         In iOS, the value is nil when the group identifier is invalid. In macOS, a URL of the expected form is always
-         returned, even if the app group is invalid, so be sure to test that you can access the underlying directory
-         before attempting to use it.
-
-         MacOS: The system also creates the Library/Application Support, Library/Caches, and Library/Preferences
-         subdirectories inside the group directory the first time you use it
-         iOS: The system creates only the Library/Caches subdirectory automatically
-
-          see: https://developer.apple.com/documentation/foundation/filemanager/1412643-containerurl/
-          */
-        func appGroupContainerUrl() -> URL? {
-            guard let appGroupIdentifier = configuration.appGroupIdentifier else { return nil }
-
-            let url = FileManager.default
-                .containerURL(forSecurityApplicationGroupIdentifier: appGroupIdentifier)?
-                .appendingPathComponent("Library/Application Support/")
-                .appendingPathComponent(Bundle.main.bundleIdentifier!)
-
-            if let url {
-                createDirectoryAtURLIfNeeded(url: url)
-                return directoryExists(url) ? url : nil
+        let apiDir = getBaseAppFolderUrl(from: configuration)
+            .appendingPathComponent(configuration.apiKey)
+
+        createDirectoryAtURLIfNeeded(url: apiDir)
+
+        // Migrate data from legacy location if it exists
+        let legacyUrl = getBaseAppFolderUrl(from: configuration)
+        if directoryExists(legacyUrl) {
+            do {
+                let fileManager = FileManager.default
+
+                // Migrate old files that correspond to StorageKey values
+                for storageKey in StorageKey.allCases {
+                    let legacyFileUrl = legacyUrl.appendingPathComponent(storageKey.rawValue)
+                    let newFileUrl = apiDir.appendingPathComponent(storageKey.rawValue)
+
+                    // Migrate the item and its contents if it exists
+                    try migrateItem(at: legacyFileUrl, to: newFileUrl, fileManager: fileManager)
+
+                    // Remove the legacy item after successful migration
+                    if fileManager.fileExists(atPath: legacyFileUrl.path) {
+                        try fileManager.removeItem(at: legacyFileUrl)
+                    }
+                }
+            } catch {
+                hedgeLog("Error during storage migration: \(error)")
```

Comment:
Lets add the path as part of the error as well, so in case things go wrong, and people share the logging, we might spot something, unless the `error` is clear about the paths